### DISCUSSION TITLE
Is it possible to setup `ignore_dirs` to be an exact match instead of the directory and its children?

### DISCUSSION THREAD

**Discussion by miversen33** [olimorris/persisted.nvim#98] [Q&A] [answered]
Basically, if I open neovim, I want it to pickup the session where I left off. But it would be nice if I could disable sessioning on specific directories (such as `$HOME`, `/tmp/`, `/`, etc). Currently if I set `$HOME` in ignored_dirs, it certainly does ignore `$HOME`, but also all things under it. Which is not great. I suspect the answer to my question is "no you cannot do that", to which my follow up question is, are you open to a PR that would provide that functionality?

Thanks for the great plugin!

**Comment by olimorris** [comment, marked as answer]
Always open to a PR and unfortunately, you're correct.

My concern with this change is that we're potentially forcing current users of the plugin to add extra granularity to their configuration by many specifying sub-folders. I take your point regarding `$HOME` but view that as an unlikely use case for the majority of users.

**Reply by miversen33** [reply]
I'll poke the code later to see how hard it would be to do this. I also maintain Neo-tree and Netman so I fully understand the desire to keep stuff "simple" as default. What I am thinking (off the cuff) is something like this

```lua
ignore_dirs = {
    "/dir1", -- Omits dir1 and its children, just like the default settings
    "/dir2", -- Omits dir2 and its children, just like the default settings
   { "/home", exact = true } -- Omits only /home but not its children
}
```

This would allow us to provide more granularity __if we want to__ but doesn't change the default at all. Basically makes me happy while not messing with the normal user.

**Reply by olimorris** [reply]
> ```lua
> 
> ignore_dirs = {
> 
>     "/dir1", -- Omits dir1 and its children, just like the default settings
> 
>     "/dir2", -- Omits dir2 and its children, just like the default settings
> 
>    { "/home", exact = true } -- Omits only /home but not its children
> 
> }
> 
> ```

Great thinking and this would be an awesome addition.
